src/OpenTelemetry.Instrumentation.AspNetCore/AspNetCoreInstrumentation.cs

```diff
@@ -22,6 +22,16 @@ namespace OpenTelemetry.Instrumentation.AspNetCore
     /// </summary>
     internal sealed class AspNetCoreInstrumentation : IDisposable
     {
+#if NET6_0_OR_GREATER
+        private static readonly HashSet<string> DiagnosticSourceEvents = new()
```

Comment:
How about using creating this HashSet for every target and then adding this?
```csharp
#!if NET6_0_OR_GREATER
    DiagnosticSourceEvents.Add("Microsoft.AspNetCore.Mvc.BeforeAction")
#endif
```

Reply:
changed - let me know if that works